container/snapshot.go

```diff
@@ -0,0 +1,152 @@
+package container
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/go-connections/nat"
+)
+
+// Snapshot is a read only view for Containers
```

Comment:
I'll add a comment about the structure.

Re: using `types.Container`, here's a brief discussion about it we had on a commit that was previously on this PR: https://github.com/docker/docker/pull/31273#discussion_r102825454